testbed/coav-sim.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SCRIPT_DIR=$(dirname $(realpath ${BASH_SOURCE[0]}))
+
+# Supported Autopilots
+AP_PX4=0
+AP_APM=1
+
+# ArduCopter Variables
+# APM_DIR will be added to PATH if set
+APM_CMD="arducopter"
+APM_TCP_PORT_1=5760
+APM_TCP_PORT_2=5762
+
+# PX4 Variables
+PX4_DIR=${PX4_DIR:-"~/px4/Firmware"}
+PX4_CMD="make posix_sitl_default jmavsim"
+PX4_UDP_PORT_1=14550
+PX4_UDP_PORT_2=14540
+
+# Simulation Parameters
+WORLD=${WORLD:-"simple_obstacle.sdf"}
+
+# Collision Avoidance and Gazebo variables
+GZSITL_UDP_PORT=15556
+COAV_GCS_UDP_PORT=15557
+
+# Set autopilot
+AUTOPILOT=${AUTOPILOT:-AP_PX4}
+
+run_autopilot () {
+        # Run SITL Simulator
+    if (("$AUTOPILOT" == "$AP_PX4")); then
```

Comment:
I need to test again, but both setting home position was not a problem. The problem was that jmavsim would stop getting data from gyro and baro, apparently because the mainloop was taking longer than it should and the sensors would timeout. That generated that crazy behavior. Try closing everything else in the machine and just run the simulation. Here, is chrome was running, the problem would appear even before takeoff. Same if qgroundcontrol was running. Try running jmavsim + qgroundcontrol without any of our code and see if it works for you (for me it doesn't!)